apps/els_lsp/src/els_dt_document.erl

```diff
@@ -204,3 +205,14 @@ wrapping_functions(Document, Line, Column) ->
 wrapping_functions(Document, Range) ->
   #{start := #{character := Character, line := Line}} = Range,
   wrapping_functions(Document, Line, Character).
+
+-spec find_candidates(binary()) -> [uri()].
+find_candidates(Pattern) ->
+  All = ets:tab2list(name()),
+  Fun = fun(#els_dt_document{uri = Uri, text = Text}) ->
```

Comment:
Does that mean we read all files of the project into memory? What impact does this have on memory and indexing time?

How does `binary:match` compare to `grep` performance wise? (Seems it should be quicker since the file IO is already done)

Reply:
That's unchanged. The loading time is minimal and not a concern (I will try to get some actual numbers).
At this point, the `match` performance is not an issue, the part that takes time is the indexing of a potentially big number of candidates (eg if you search for callers to a function called "get").

Reply:
By implementing support for `WorkDoneProgress` we can at least report progress to the user and hopefully show intermediate results and be able to cancel a specific request.